src/main/java/software/amazon/cloudformation/proxy/AmazonWebServicesClientProxy.java

```diff
@@ -110,6 +122,70 @@ public ClientT client() {
         };
     }
 
+    public <ClientT, ModelT, CallbackT extends StdCallbackContext>
+        Initiator<ClientT, ModelT, CallbackT>
+        newInitiator(@Nonnull Supplier<ClientT> client, final ModelT model, final CallbackT context) {
+        return newInitiator(newProxy(client), model, context);
+    }
+
+    @Override
+    public <ClientT, ModelT, CallbackT extends StdCallbackContext>
+        Initiator<ClientT, ModelT, CallbackT>
+        newInitiator(final ProxyClient<ClientT> client, final ModelT model, final CallbackT context) {
+        Preconditions.checkNotNull(client, "ProxyClient can not be null");
+        Preconditions.checkNotNull(model, "Resource Model can not be null");
+        Preconditions.checkNotNull(context, "cxt can not be null");
+        return new StdInitiator<>(client, model, context);
+    }
+
+    private class StdInitiator<ClientT, ModelT, CallbackT extends StdCallbackContext>
+        implements Initiator<ClientT, ModelT, CallbackT> {
+
+        private final ProxyClient<ClientT> client;
+        private final ModelT model;
+        private final CallbackT callback;
+
+        private StdInitiator(final ProxyClient<ClientT> client,
+                             final ModelT model,
+                             final CallbackT callback) {
+            Preconditions.checkNotNull(client, "ProxyClient can not be null");
+            Preconditions.checkNotNull(model, "Resource Model can not be null");
+            Preconditions.checkNotNull(callback, "cxt can not be null");
+            this.client = client;
+            this.model = model;
+            this.callback = callback;
+        }
+
+        @Override
+        public RequestMaker<ClientT, ModelT, CallbackT> initiate(String callGraph) {
+            return new CallContext<>(callGraph, client, model, callback);
+        }
+
+        @Override
+        public ModelT getResourceModel() {
+            return model;
+        }
+
+        @Override
+        public CallbackT getCallbackContext() {
+            return callback;
+        }
+
+        @Override
+        public <NewModelT> Initiator<ClientT, NewModelT, CallbackT> rebindModel(NewModelT model) {
```

Comment:
What is this method for?

Reply:
There are cases where I want to rebind just the model but continue to use the client and the callback context. As an example 

```java
    protected void addServiceAccess(String serviceName, String kmsKeyId, Region region) {
        final CallChain.Initiator<KmsClient, ObjectNode, StdCallbackContext> initiator =
            this.initiator.rebindModel(mapper.createObjectNode());
  // use ObjectNode for model
  getKeyPolicyAddServiceAccess(initiator, serviceName, region) 
            .then(event -> {
                ObjectNode policy = event.getResourceModel();
                ProgressEvent<String, StdCallbackContext> progress = 
                    // rebind model to string to convey policy document
                    initiator.rebindModel(inline)  // rebind model to string for policy doc
                    .initiate("kms:PutKeyPolicy")
                    .translate(policyDoc -> PutKeyPolicyRequest.builder()
                        .keyId(kmsKeyId)
                        .policyName("default")
                        .policy(policyDoc)
                        .build())
                 .backoff(Constant.of().delay(Duration.ofSeconds(10)).timeout(Duration.ofSeconds(5 * 10)).build())
                    .call((r, c) -> c.injectCredentialsAndInvokeV2(r, c.client()::putKeyPolicy))
                    .success();
```

In the above code we switch the model to an ObjectNode. This code retrieves the KMS keys' associated IAM policy as a JSON object using ObjectNode, adds a policy object for service access and updates it using kms:PutKeyPolicy.